### PR TITLE
deps: use platform-go-middlewares v2

### PR DESCRIPTION
**Pull request by ezr-ondrej**
# Description
Migrate to identity v2 for better support of service accounts.

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [x] Refactor

Needs https://github.com/RedHatInsights/edge-api/pull/2457

### CODE REVIEW

**Comment by ezr-ondrej** [comment]
@lzap @loadtheaccumulator could you take a peek? :)

File: logger/logger.go [hunk 1]
```diff
@@ -42,10 +42,11 @@ func InitLogger(writer io.Writer) {
 	if cfg.Logging != nil && cfg.Logging.Region != "" {
 		cred := credentials.NewStaticCredentials(cfg.Logging.AccessKeyID, cfg.Logging.SecretAccessKey, "")
 		awsconf := aws.NewConfig().WithRegion(cfg.Logging.Region).WithCredentials(cred)
-		hook, err := lc.NewBatchingHook(cfg.Logging.LogGroup, cfg.Hostname, awsconf, 10*time.Second)
+		w, err := lc.NewBatchWriterWithDuration(cfg.Logging.LogGroup, cfg.Hostname, awsconf, 10*time.Second)
 		if err != nil {
 			log.WithFields(log.Fields{"error": err.Error()}).Error("Error creating AWS hook")
 		}
+		hook = lc.NewLogrusHook(w)
```
Comment (lzap) [inline logger/logger.go:49]: Why this change? What does it has to do with identity?

**Review by lzap — DISMISSED** [review]
I see no problems, however, I am currently unable to test this. I do not have a fully development setup yet. This can possibly break the app completely. Needs to be tested first.

File: cmd/migrategroups/migrategroups/migrategroups.go [hunk 2]
```diff
@@ -77,15 +75,14 @@ func newInventoryOrgClients(orgID string) (*InventoryOrgClients, error) {
...
-		User:     identity.User{OrgAdmin: true, Email: "edge-groups-migrator@example.com", FirstName: "edge-groups-migrator"},
+		User:     &identity.User{OrgAdmin: true, Email: "edge-groups-migrator@example.com", FirstName: "edge-groups-migrator"},
 	}}
 	jsonIdent, err := json.Marshal(&ident)
 	if err != nil {
 		return nil, err
 	}
 	base64Identity := base64.StdEncoding.EncodeToString(jsonIdent)
 	ctx := context.Background()
-	ctx = context.WithValue(ctx, request_id.RequestIDKey, uuid.NewString())
```
[the remark this answers is not in the document]
Comment (ezr-ondrej) [reply cmd/migrategroups/migrategroups/migrategroups.go:88]: it's just adding a request_id to the client headers, but IMO it's useless, because we are not storing it, so it's generated and not very useful (I'm open to ideas, but I felt this is the easiest way).

**Comment by lzap** [comment]
Just for the record, I am working on a Gorm context improvements where I actually use the context request:

```
func (l *GormLogger) Error(ctx context.Context, msg string, data ...interface{}) {
	rid := request_id.GetReqID(ctx)
	l.logger.WithFields(logrus.Fields{
		"request_id": rid,
	}).Errorf(msg, data...)
}
```

@loadtheaccumulator I am bit hesitant to merge this as I don’t know how to properly test things yet

**Comment by ezr-ondrej** [comment]
lets draft it for now ;)

**Comment by lzap** [comment]
The patch is great, don’t get me wrong, I am just a bit worried to merge this right now. When you will be rebasing, make sure to store request_id in the context so it can be retrieved by the Gorm logger.

File: cmd/migrategroups/migrategroups/migrategroups.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (lzap) [reply cmd/migrategroups/migrategroups/migrategroups.go:88]: Just to be sure - I added `request_id.RequestIDKey` in my other PR at few places, make sure to refactor that code as well.

File: cmd/migraterepos/migraterepos/migraterepos.go [hunk 3]
```diff
@@ -26,15 +25,14 @@ func newOrgContentSourcesClient(orgID string) (repositories.ClientInterface, err
...
-		User:     identity.User{OrgAdmin: true, Username: "edge-repo-migrator"},
+		User:     &identity.User{OrgAdmin: true, Username: "edge-repo-migrator"},
 	}}
 	jsonIdent, err := json.Marshal(&ident)
 	if err != nil {
 		return nil, err
 	}
 	base64Identity := base64.StdEncoding.EncodeToString(jsonIdent)
 	ctx := context.Background()
-	ctx = context.WithValue(ctx, request_id.RequestIDKey, uuid.NewString())
```
Comment (lzap) [inline cmd/migraterepos/migraterepos/migraterepos.go:37]: You removed this because it made no sense, correct?

Comment (ezr-ondrej) [reply cmd/migraterepos/migraterepos/migraterepos.go:37]: Yes, correct. I do not see a reason to generate a request id and save it nowhere.
Also this way no longer works

File: cmd/migrategroups/migrategroups/migrategroups.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ezr-ondrej) [reply cmd/migrategroups/migrategroups/migrategroups.go:88]: hmm I did not find any :thinking:

File: pkg/routes/common/identity.go [hunk 4]
```diff
@@ -53,3 +49,25 @@ func GetIdentityInstanceFromContext(ctx context.Context) (identity.XRHID, error)
...
+	switch id.Identity.Type {
+	case "User":
+		return id.Identity.User.Username
+	case "ServiceAccount":
+		return id.Identity.ServiceAccount.Username
+	case "System":
+		return id.Identity.System.CommonName
+	}
+	return ""
+}
```
Comment (lzap) [inline pkg/routes/common/identity.go:73]: What this is introduced for?

Comment (ezr-ondrej) [reply pkg/routes/common/identity.go:73]: To FIX this: https://github.com/RedHatInsights/edge-api/blob/09a1973e5e04d9a4076ce796e5f976efb4783ebd/pkg/services/updates.go#L292

Addmitedly it could have been a follow up PR.

**Review by lzap — APPROVED** [review]
I do not see any issues, it is a dangerous patch given that we took ownership of this repository weeks ago. We need to trust the IQE tests at this point and do additional testing on stage.
